packages/ui-stencil/src/components/orama-chat-box/orama-chat-box.tsx

```diff
@@ -55,23 +59,54 @@ export class ChatBox {
   chatMarkdownLinkClicked: EventEmitter<OnChatMarkdownLinkClickedCallbackProps>
 
   @Watch('index')
-  indexChanged() {
-    this.startChatService()
+  async indexChanged(newValue: CloudIndexConfig) {
+    if (newValue) {
+      await this.startChatService();
+    }
   }
 
-  componentWillLoad() {
-    this.el.id = this.componentID
-    this.startChatService()
+  async componentWillLoad() {
+    this.el.id = this.componentID;
+    if (this.index || this.clientInstance) {
+      await this.startChatService();
+    }
+  }
+
+  async componentDidLoad() {
+    if (!chatContext.chatService && (this.index || this.clientInstance)) {
+      await this.startChatService();
+    }
   }
 
-  startChatService() {
-    validateCloudIndexConfig(this.el, this.index, this.clientInstance)
-    const oramaClient = this.clientInstance || initOramaClient(this.index)
 
-    chatContext.chatService = new ChatService(oramaClient)
+  async startChatService() {
+    if (this.initializationAttempts >= this.MAX_ATTEMPTS) {
+      return;
+    }
+
+    try {
+      if (!this.index && !this.clientInstance) {
+        return;
+      }
+
+      validateCloudIndexConfig(this.el, this.index, this.clientInstance);
+      const oramaClient = this.clientInstance || await initOramaClient(this.index);
+      chatContext.chatService = new ChatService(oramaClient);
+      this.isServiceReady = true;
+
+      this.el.dispatchEvent(new CustomEvent('chatServiceReady'));
```

Comment:
Is there any particular reason for not use stencil events?

https://stenciljs.com/docs/events